### PR TITLE
running extension for chrome

### PR DESCRIPTION
**Pull request by woolfg**

### CODE REVIEW

**Comment by woolfg** [comment]
already submitted for review to google chrome store

File: package.json [hunk 1]
```diff
@@ -5,9 +5,11 @@
   "license": "MIT",
   "private": true,
   "scripts": {
-    "start:firefox": "web-ext run --source-dir ./src/"
+    "start:firefox": "web-ext run --source-dir ./src/",
```
Comment (mre) [inline package.json:8]: Is there a `-t firefox`, just like for Chromium? If so, maybe we should be explicit here.

File: src/background.js [hunk 2]
```diff
@@ -1,24 +1,20 @@
-let namespace = window.browser || chrome
+const runtime = chrome || window.browser
+
+//register handler for communication with content script
+runtime.runtime.onMessage.addListener((request, sender, sendResponse) => {
```
Comment (mre) [inline src/background.js:4]: `runtime.runtime` sounds weird. Maybe we should call it `browser`.

File: src/background.js [hunk 3]
```diff
@@ -1,24 +1,20 @@
...
+
+  const runtime = chrome || window.browser
```
Comment (mre) [inline src/background.js:6]: why was that defined twice?

File: src/manifest.json [hunk 4]
```diff
@@ -1,23 +1,26 @@
 {
   "name": "OpenPodcast Extension",
   "version": "1.0",
-  "description": "Retrieve needed keys and credentials for Spotify and Apple connector",
-  "browser_action": {
-    "default_popup": "popup.html"
-  },
+  "description": "Shows Spotify keys on openpodcast.dev to connect Podcasters Dashboard with Open Podcast Analytics",
```
Comment (mre) [inline src/manifest.json]:
```suggestion
  "description": "Shows Spotify keys on openpodcast.dev to connect a Podcasters Dashboard with Open Podcast Analytics",
```

File: src/manifest.v2.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "OpenPodcast Extension",
+  "version": "1.0",
+  "description": "Shows Spotify keys on openpodcast.dev to connect Podcasters Dashboard with Open Podcast Analytics",
```
Comment (mre) [inline src/manifest.v2.json]:
```suggestion
  "description": "Shows Spotify keys on openpodcast.dev to connect a Podcasters Dashboard with Open Podcast Analytics",
```

File: src/openpodcast.js [hunk 6]
```diff
@@ -1,25 +1,39 @@
+const runtime = window.browser || chrome
+
 //insert html at the beginning of the website
 function showCredentialsOnWebsite(creds) {
-  let message = "No Spotify credentials found, are you logged in on podcasters.spotify.com? (reload needed)"
+  let message = "No Spotify credentials found, are you logged in on podcasters.spotify.com? (maybe reload this page)"
```
Comment (mre) [inline src/openpodcast.js]:
```suggestion
  let message = "No Spotify credentials found, are you logged in on podcasters.spotify.com? (Maybe reload this page.)"
```

File: src/openpodcast.js [hunk 7]
```diff
@@ -1,25 +1,39 @@
...
   if (creds.sp_key && creds.sp_dc) {
-    message = `Spotify Credentials: key: ${creds['sp_key']}, dc: ${creds['sp_dc']}`
+    message = `Spotify keys: key: ${creds['sp_key']}, dc: ${creds['sp_dc']} END`
```
Comment (mre) [inline src/openpodcast.js]: `keys: key` can be confusing. How about `Spotify cookies: key`?

File: src/openpodcast.js [hunk 8]
```diff
@@ -1,25 +1,39 @@
...
   }
-  const credsHTMLElement = document.createTextNode(message);
-  document.body.prepend(credsHTMLElement)
+  document.getElementById("openpodcast-plugin").innerHTML = message
 }
 
 //communicate with background script to fetch cookies (only background script is allowed to do this)
 async function retrieveSpotifyCookies() {
+  console.log("retrieving spotify keys from service worker")
```
Comment (mre) [inline src/openpodcast.js:14]: Do you want to keep this line or is it debug output?

File: src/openpodcast.js [hunk 9]
```diff
@@ -1,25 +1,39 @@
...
   // sends a message to the background.js and retrieve cookies
-  const message = await browser.runtime.sendMessage({type: "spotifycookies"})
+
+  const message = await chrome.runtime.sendMessage({ type: "spotifycookies" })
+
   const cookies = message.response
+  console.log(cookies)
```
Comment (mre) [inline src/openpodcast.js]: same here

File: src/openpodcast.js [hunk 10]
```diff
@@ -1,25 +1,39 @@
...
 
-//init
-retrieveSpotifyCookies()
\ No newline at end of file
+// check if current website is the OpenPodcast website where we want to show the credentials
+function isOnOpenPodcastWebsite() {
+  return window.location.href.match(/https:\/\/openpodcast.dev\/.*connect/) && document.getElementById("openpodcast-plugin")
+}
+
+//wait 1sec until the website is loaded (maybe a problem with docsify) and then check if we are on the right website
```
Comment (mre) [inline src/openpodcast.js:33]: docsify can take multiple seconds to load from my experience. Maybe we want to increase the waiting time to 2-3 seconds?

**Review by mre — APPROVED** [review]
Just some minor nitpicks. Can be merged from my side.

File: src/background.js (hunk 2 above)
Comment (woolfg) [reply src/background.js:4]: browser is protected. `runtime` is used in many docs and examples, that is why I used it.

File: src/background.js (hunk 3 above)
Comment (woolfg) [reply src/background.js:6]: behaviour of service workers, the outer const is removed over time. drove me crazy....

File: src/openpodcast.js (hunk 7 above)
Comment (woolfg) [reply src/openpodcast.js]: we have to clean this later anyways. cookie is also not well understood by people. for now it should be fine, I also didn't want to trigger the reviewer by reading "cookie" information.

File: src/openpodcast.js (hunk 10 above)
Comment (woolfg) [reply src/openpodcast.js:33]: it is a loop anyways

**Pull request merged**